crates/common/tedge_utils/src/notify.rs

```diff
@@ -0,0 +1,393 @@
+
+            if maybe_file_name_entry.contains_key(&Some(file_name.clone())) {
+                let events = maybe_file_name_entry.get(&Some(file_name)).unwrap();
+                return events.contains(event);
+            } else if maybe_file_name_entry.contains_key(&None) {
+                let events = maybe_file_name_entry.get(&None).unwrap();
+                return events.contains(event);
+            }
+        }
+        false
+    }
+}
+
+pub struct NotifyStream {
+    watcher: INotifyWatcher,
+    pub rx: Receiver<(PathBuf, FileEvent)>,
+    metadata: WatchMetadata,
+}
+
+impl TryDefault for NotifyStream {
+    type Error = NotifyStreamError;
+
+    fn try_default() -> Result<Self, Self::Error> {
+        let (tx, rx) = channel(1024);
+
+        // Automatically select the best implementation for your platform.
+        // You can also access each implementation directly e.g. INotifyWatcher.
+        let watcher = RecommendedWatcher::new(
+            move |res: Result<notify::Event, notify::Error>| {
+                futures::executor::block_on(async {
+                    if let Ok(notify_event) = res {
+                        match notify_event.kind {
+                            EventKind::Modify(ModifyKind::Data(DataChange::Any)) => {
+                                for path in notify_event.paths {
+                                    tx.send((path, FileEvent::Modified)).await.unwrap();
+                                }
+                            }
+
+                            EventKind::Access(AccessKind::Close(AccessMode::Write)) => {
+                                for path in notify_event.paths {
+                                    tx.send((path, FileEvent::Modified)).await.unwrap();
+                                }
+                            }
+                            EventKind::Create(CreateKind::File) => {
+                                for path in notify_event.paths {
+                                    tx.send((path, FileEvent::Created)).await.unwrap();
+                                }
+                            }
+                            EventKind::Create(CreateKind::Folder) => {
+                                for path in notify_event.paths {
+                                    tx.send((path, FileEvent::Created)).await.unwrap();
+                                }
+                            }
+                            //EventKind::Modify(_) => todo!(),
+                            EventKind::Remove(RemoveKind::File) => {
+                                for path in notify_event.paths {
+                                    tx.send((path, FileEvent::Deleted)).await.unwrap();
+                                }
+                            }
+                            EventKind::Remove(RemoveKind::Folder) => {
+                                for path in notify_event.paths {
+                                    tx.send((path, FileEvent::Deleted)).await.unwrap();
```

Comment:
All the above `unwrap`s have to be handled properly.

Reply:
Sometimes these errors do happen but the tests do not fail. I have not yet looked into how to better handle them.

Reply:
Handled. Thank you